### PR TITLE
Add check of vacation start day

### LINKED ISSUE #3: Need check vacation start date in user request
User may request vacation not early than in 2 weeks (Considering request day).
If interval between today's day and vacation start date smaller than 2 weeks then bot must reject it.

### PR DESCRIPTION
**Pull request by polina-popova**
Closes #3

### CODE REVIEW

File: src/viva.js [hunk 1]
```diff
@@ -142,12 +145,34 @@ module.exports = function (robot) {
     }
 
     if (state.n === FROM_STATE) {
+      const today = moment()
+      let year
+
+      if (today.month()+1 > parseInt(month)) {
+        year = moment().year() + 1
+      } else if (today.month()+1 < parseInt(month, 10)) {
```
Comment (ukhryab) [inline src/viva.js]: Please format your code properly.

**Comment by ukhryab** [comment]
Tested, it works.

File: src/viva.js [hunk 2]
```diff
@@ -20,6 +20,7 @@ module.exports = function (robot) {
   const MAXIMUM_LENGTH_OF_LEAVE = parseInt(process.env.MAXIMUM_LENGTH_OF_LEAVE, 10) || 28
   const MAXIMUM_LENGTH_OF_WAIT = parseInt(process.env.MAXIMUM_LENGTH_OF_WAIT, 10) || 7
   const REMINDER_SCHEDULER = process.env.REMINDER_SCHEDULER || '0 0 7 * * *'
+  const MINIMUM_DAYS_BEFORE_ASK = parseInt(process.env.MINIMUM_DAYS_BEFORE_ASK, 14) || 14
```
Comment (eugulixes) [inline src/viva.js]: * The script deals with leave requests, so I suggest we will name the constant `MINIMUM_DAYS_BEFORE_REQUEST`.
* Note that the constants are sorted in the alphabetic order.

File: src/viva.js [hunk 3]
```diff
@@ -142,11 +145,22 @@ module.exports = function (robot) {
     }
 
     if (state.n === FROM_STATE) {
+      const today = moment()
+      const year = today.month() + 1 >= month && today.date() >= day ? today.year() + 1 : today.year()
```
Comment (eugulixes) [inline src/viva.js]: Let's put a comment here that `month` starts from 0.

File: src/viva.js [hunk 4]
```diff
@@ -142,11 +145,23 @@ module.exports = function (robot) {
     }
 
     if (state.n === FROM_STATE) {
+      const today = moment() 
+      //moment().month() start count from 0
```
Comment (eugulixes) [inline src/viva.js]: Fix: ```moment().month() starts counting with zero```.

File: src/viva.js [hunk 5]
```diff
@@ -142,11 +145,23 @@ module.exports = function (robot) {
     }
 
     if (state.n === FROM_STATE) {
+      const today = moment()
+      // moment().month() start count from 0
```
Comment (eugulixes) [inline src/viva.js]: Unfortunately, it doesn't sound right. Fixed variant: ```moment().month() starts counting with 0```.

File: src/viva.js [hunk 6]
```diff
@@ -142,11 +145,23 @@ module.exports = function (robot) {
     }
 
     if (state.n === FROM_STATE) {
+      const today = moment()
+      // moment().month() starts count with 0
```
Comment (eugulixes) [inline src/viva.js]: `counting` would be better.

**Comment by BehindLoader** [comment]
I Tested. Everything is fine

**Pull request merged**